ossrelease/gen_man.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+'''
+Generates the man pages for a given release
+'''
+
+import argparse
+import re
+import shutil
+import subprocess
+import os
+
+# import ossrelease modules
+import conf
+
+def main():
+    '''
+    Run!
+    Prints results to the screen.
+    '''
+    # Parse args and define some basic params
+    opts = conf.get_conf()
+    args = parse_args()
+    doc_dir = opts['SALT_REPO_PATH'] + '/doc/'
+    man_build_dir = doc_dir + '_build/man/'
+    man_dir = doc_dir + 'man/'
+    branch = '.'.join(args.version.split('.')[:-1])
+
+    print('Building man pages in directory: {0}'.format(doc_dir))
+
+    # Check out master branch
+    _cmd_run(['make', 'man', '-C', doc_dir])
+
+    print('Copying new man files from {0} to {1}'.format(man_build_dir,
+                                                         man_dir))
+    for file_ in os.listdir(man_build_dir):
+        file_path = man_build_dir + file_
+        print('Copying file: {0}'.format(file_path))
+        shutil.copy(file_path, man_dir)
+
+    for file_ in os.listdir(man_dir):
+        _replace_txt(man_dir + file_, old='"{0}.*" '.format(branch),
+                     new='"{0}" '.format(args.version), regex=True)
+
+        print('Adding Salt Version {0} to file: {1}'.format(args.version,
+
+def _replace_txt(file_name, old=None, new=None, regex=False):
+    with open(file_name) as fh_:
+        file_txt = fh_.read()
+    if regex:
+        find = re.search(old, file_txt)
+        if not find:
+            print('{0} not found in file: {1}'.format(old, file_name))
+            return False
+        else:
+            old = find.group(0)
+
+    with open(file_name, 'w') as fh_:
+        fh_.write(file_txt.replace(old, new))
+
+def parse_args():
+    '''
+    Parse the CLI options.
+    '''
+    # Define parser and set up basic options
+    parser = argparse.ArgumentParser(
+        description='Update the salt man pages'
+    )
+    parser.add_argument('-v', '--version',
```

Comment:
I am thinking this argument might be named something else. Like `branch`? When I saw `-v` I thought immediately that this would either mean to print logs in `verbose` or list the `version` of the script. Not "pass the version of salt to build man pages for". 

I know that is picky, but I am just thinking about this script moving forward and how it might tie in with other features or could be confusing with the `-v` option. Definitely open to other opinions or rebuttals. :)

Reply:
what about just including `--version` the long hand part instead. Branch doesn't really make sense because we are defining the salt version not branch.

Reply:
sure that is fine.